lib/Controller/ApiGeneralController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/** @noinspection DuplicatedCode */
+
+namespace OCA\Tables\Controller;
+
+use Exception;
+use OCA\Tables\AppInfo\Application;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\ResponseDefinitions;
+use OCA\Tables\Service\TableService;
+use OCA\Tables\Service\ViewService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @psalm-import-type TablesIndex from ResponseDefinitions
+ */
+class ApiGeneralController extends OCSController {
+	private TableService $tableService;
+	private ViewService $viewService;
+
+	private ?string $userId;
+	protected LoggerInterface $logger;
+
+	use Errors;
+
+
+	public function __construct(
+		IRequest     $request,
+		TableService $tableService,
+		ViewService $viewService,
+		LoggerInterface $logger,
+		?string $userId
+	) {
+		parent::__construct(Application::APP_ID, $request);
+		$this->tableService = $tableService;
+		$this->viewService = $viewService;
+		$this->userId = $userId;
+		$this->logger = $logger;
+	}
+
+
+	/**
+	 * [api v2] Returns all main resources
+	 *
+	 * Tables and views incl. shares
+	 *
+	 * @NoAdminRequired
+	 * @CORS
+	 * @NoCSRFRequired
+	 *
+	 * @return DataResponse<Http::STATUS_OK, TablesIndex, array{}>|DataResponse<Http::STATUS_INTERNAL_SERVER_ERROR, array{message: string}, array{}>
+	 *
+	 * 200: Index returned
+	 */
+	public function initIndex(): DataResponse {
```

Comment:
```suggestion
	public function index(): DataResponse {
```

I found the init a bit confusing as it doesn't initialize?